### PR TITLE
Enable custom Rails Database Connection Timeout

### PR DESCRIPTION
**Pull request by pablo-code-org**
This increases the timeout when creating a database (and indexing) using an environment variable..

This will affect all environments but only when doing db:setup_migrate

Note: I am still not sure if I chose the right parameter to increase (db_read_timeout)

This is done because the DB is timing out while create a [new index](https://github.com/code-dot-org/code-dot-org/pull/49918)

```
buntu@production-daemon:~/production/dashboard$ RAILS_ENV=production RACK_ENV=production bundle exec rake db:setup_or_migrate --trace
GetSecretValue: production/cdo/slack_token
GetSecretValue: production/cdo/slack_bot_token
GetSecretValue: production/cdo/mapbox_access_token
** Invoke db:setup_or_migrate (first_time)
** Execute db:setup_or_migrate
** Invoke environment (first_time)
** Execute environment
GetSecretValue: production/cdo/dashboard_devise_secret
GetSecretValue: production/cdo/dashboard_devise_pepper
GetSecretValue: production/cdo/dashboard_facebook_key
GetSecretValue: production/cdo/dashboard_facebook_secret
GetSecretValue: production/cdo/dashboard_google_key
GetSecretValue: production/cdo/dashboard_google_secret
GetSecretValue: production/cdo/dashboard_windowslive_key
GetSecretValue: production/cdo/dashboard_windowslive_secret
GetSecretValue: production/cdo/dashboard_clever_key
GetSecretValue: production/cdo/dashboard_clever_secret
GetSecretValue: production/cdo/dashboard_schoolproject_key
GetSecretValue: production/cdo/dashboard_schoolproject_secret
GetSecretValue: production/cdo/recaptcha_site_key
GetSecretValue: production/cdo/recaptcha_secret_key
GetSecretValue: production/cdo/dashboard_secret_key_base
GetSecretValue: production/cdo/zendesk_dev_token
GetSecretValue: production/cdo/dashboard_honeybadger_api_key
Ignoring db/schema_cache.yml because it has expired. The current schema version is 20221024233909, but the one in the cache is 20230124093901.
** Invoke db:load_config (first_time)
** Invoke environment 
** Execute db:load_config
** Invoke db:migrate (first_time)
** Invoke db:load_config 
** Execute db:migrate
== 20230124093901 CreateProjectStateAndStorageIdIndex: migrating ==============
-- add_index(:projects, [:storage_id, :state], {:name=>"storage_apps_storage_id_state_index"})
rake aborted!
StandardError: An error has occurred, all later migrations canceled:

Mysql2::Error::TimeoutError: Timeout waiting for a response from the last query. (waited 30 seconds)
```

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/lib/tasks/setup_or_migrate.rake [hunk 1]
```diff
@@ -14,6 +14,7 @@ namespace :db do
 
   timed_task_with_logging :setup_or_migrate do
     db_exists = database_exists?
+    ENV['db_read_timeout'] = 300
```
Comment (sureshc) [inline dashboard/lib/tasks/setup_or_migrate.rake]: Might be useful to adopt our convention of using all caps for environment variable names (`DB_READ_TIMEOUT`).

File: dashboard/config/database.yml [hunk 2]
```diff
@@ -39,7 +39,7 @@ mysql_defaults: &mysql_defaults
   # You can set the value so that a lost connection can be detected earlier than the TCP/IP Close_Wait_Timeout value of 10 minutes.
   #
   # NOTE: mysql2 reuses this variable as a query timeout without retry, so set it to the maximum query execution time.
-  read_timeout: 30
+  read_timeout:  <%= unless ENV['db_read_timeout'].nil && ENV['db_read_timeout'] > 0 ? ENV['db_read_timeout']   : 30 %>
```
Comment (sureshc) [inline dashboard/config/database.yml]: We might be able to do something like:

``` ruby
read_timeout: ENV['DB_READ_TIMEOUT'] || 30
```

Comment (pablo-code-org) [reply dashboard/config/database.yml]: I didn't know which one would be the right syntax :p 

Thanks!

Comment (pablo-code-org) [reply dashboard/config/database.yml]: I wasn't sure if the lack of value is nil or zero. Because if it is zero it would make it have no limit

**Pull request closed without merging**